packages/react-components/lib/tasks/create-task.tsx

```diff
@@ -1581,6 +649,69 @@ export function CreateTaskForm({
     }
   };
 
+  // Order the menu items based on the supported tasks
+  const renderMenuItem = (taskName: string, taskNameRemap: Record<string, string>) => {
+    switch (taskName) {
+      case 'patrol':
+        return (
+          <MenuItem value={taskName} disabled={!patrolWaypoints || patrolWaypoints.length === 0}>
+            {taskNameRemap[taskName] ?? 'Patrol'}
+          </MenuItem>
+        );
+      case 'delivery':
+        return (
+          <MenuItem
+            value={taskName}
+            disabled={
+              Object.keys(pickupPoints).length === 0 || Object.keys(dropoffPoints).length === 0
+            }
+          >
+            {taskNameRemap[taskName] ?? 'Delivery'}
+          </MenuItem>
+        );
+      case 'clean':
+        return (
+          <MenuItem value={taskName} disabled={!cleaningZones || cleaningZones.length === 0}>
+            {taskNameRemap[taskName] ?? 'Clean'}
+          </MenuItem>
+        );
+      case 'custom_compose':
+        return (
+          <MenuItem value={taskName}>{taskNameRemap[taskName] ?? 'Custom Compose Task'}</MenuItem>
+        );
+      case 'delivery_pickup':
+        return (
+          <MenuItem
+            value={taskName}
+            disabled={
+              Object.keys(pickupPoints).length === 0 || Object.keys(dropoffPoints).length === 0
+            }
+            sx={{
+              '& .MuiMenu-list': {
+                fontSize: isScreenHeightLessThan800 ? '0.8rem' : '1.15',
+              },
+            }}
+          >
+            {taskNameRemap[taskName] ?? 'Delivery - 1:1'}
+          </MenuItem>
+        );
+      case 'delivery_sequential_lot_pickup':
+        return (
+          <MenuItem value={taskName} disabled={Object.keys(dropoffPoints).length === 0}>
+            {taskNameRemap[taskName] ?? 'Delivery - Sequential lot pick up'}
+          </MenuItem>
+        );
+      case 'delivery_area_pickup':
+        return (
+          <MenuItem value={taskName} disabled={Object.keys(dropoffPoints).length === 0}>
+            {taskNameRemap[taskName] ?? 'Delivery - Area pick up'}
+          </MenuItem>
+        );
+      default:
+        return undefined;
+    }
+  };
```

Comment:
Do we really need different behavior for different category? Can we just use the same rendering for all task types?

Reply:
Yes we do, there were instances where users were confused why certain tasks did not allow any options (no cleaning zones in office world, for example), by disabling menu items whenever these tasks are invalid (according to the implementation), we can direct user's actions and prevent mistakes

Reply:
We could just remove the invalid tasks from the `supportedTasks` if we know a deployment does not a certain task. Disabling the menu items would cause a different set of ux issues as well, if we expect a deployment to support cleaning task, and if there is no valid cleaning zone for some reason, users will be confused why cleaning tasks are disabled even though it is supported. Not to mention the way it is now, with different item having different styling (font size) is also bad for ui.

Reply:
I beg to differ, it will still be the responsibility of the developer to keep in sync their building map yaml to contain the proper waypoints and properties. In the event that their deployment is meant to support a `Clean` task for example, after the building map has been changed, it should show up but not allow them to run it (because the waypoints may have been changed). This is a sign for the developer to fix what is needed

Users won't care or be touching the dashboard config or the building map yaml at all

Reply:
We can add tooltips to disabled menu items to let them know why certain menu items are disabled.

> Not to mention the way it is now, with different item having different styling (font size) is also bad for ui.

This is only temporary until we move to json form. I would rather have bad looking UI for now, than things not working

Reply:
> I beg to differ, it will still be the responsibility of the developer to keep in sync their building map yaml to contain the proper waypoints and properties. In the event that their deployment is meant to support a Clean task for example, after the building map has been changed, it should show up but not allow them to run it (because the waypoints may have been changed). This is a sign for the developer to fix what is needed

I don't understand the example, if the clean task shows up but cannot be ran, it means there is no valid waypoints. Even if we leave the menu item enabled, there would be no valid waypoints so the user still cannot submit the task.

> This is only temporary until we move to json form. I would rather have bad looking UI for now, than things not working

I would say this is more important because we are moving towards using ui schemas. We would not be able to define custom logic to enable/disable a category with the schemas, this particular menu is also selecting the "task type" or the schema to use and is outside the scope of the schemas. Because we are going to use ui schemas, this needs to be a list where we treat every menu item the same.